src/Octoshift/Commands/WaitForMigration/WaitForMigrationCommandHandler.cs

```diff
@@ -84,13 +85,15 @@ private async Task WaitForRepositoryMigration(string migrationId, GithubApi gith
             if (RepositoryMigrationStatus.IsSucceeded(state))
             {
                 _log.LogSuccess($"Migration {migrationId} succeeded for {repositoryName}");
+                _warningsCountLogger.LogWarningsCount(warningsCount);
```

Comment:
If you have code you want to reuse across multiple command handlers, then putting it into a service and injecting it like this is the right way to do it.

My main piece of feedback is this feels like a really tiny/trivial thing to build a whole service around. If it was me I'd probably ditch the service and just do this in the handlers:

```suggestion
                _log.LogWarning($"{warningsCount} warning(s) encountered during this migration");
```

Reply:
but if you do choose to leave it as a service, there should probably be a unit test or two for the new service.

Reply:
When I started working on this, I started with a private method in a handler class, and then discovered that there are four places it needed to be implemented.  I'm also using `LogInformation` when there are no warnings, and `LogWarning` when warnings are present.

It still feels like a :pinching_hand: change, and it does feel like kind of a lot for a service class.  We could move this to `OctoLogger` instead.  I decided not to put it in this class because I liked the decoupling of the logging class and the logging content, but this isn't a bad place to put it, either.

This gives us three reasonable places this could be placed:

- In four private methods as identical code
- A service class that contains common code
- A public method in OctoLogger

Any strong feelings about this?

Reply:
I don't have strong feelings on this.

Reply:
IMO adding something like this into OctoLogger is the worst of the three options. OctoLogger code is all at a pretty consistent level of "abstraction". It's all general-purpose logging code, with nothing in that class specific to octoshift (you could copy-paste OctoLogger into any .Net CLI project).

Another possible option that I'm not sure how I feel about could be adding it as an extension method to OctoLogger (extension methods are kind of .Net version of mixins). This way it's separate from the OctoLogger class to avoid the above concern. But adding extension methods to a class we wrote is a little weird, but maybe this is a good justification of it.

Reply:
> Another possible option that I'm not sure how I feel about could be adding it as an extension method to OctoLogger (extension methods are kind of .Net version of mixins)

I'll be wrapping up this PR. To come back to this suggestion.. any updated thoughts on this? @dylan-smith @synthead 

Otherwise will stick with the current approach.

Reply:
Coming back to this, I do personally like the current approach - but unit tests would be great.